junit5/src/main/java/cz/xtf/junit5/extensions/OpenShiftRecorderHandler.java

```diff
@@ -1,220 +1,54 @@
 import org.junit.jupiter.api.extension.ExtensionContext;
-import org.junit.jupiter.api.extension.ExtensionContext.Namespace;
-import org.junit.jupiter.api.extension.ExtensionContext.Store;
 import org.junit.jupiter.api.extension.LifecycleMethodExecutionExceptionHandler;
 import org.junit.jupiter.api.extension.TestExecutionExceptionHandler;
 import org.junit.jupiter.api.extension.TestWatcher;
-import org.junit.platform.commons.support.AnnotationSupport;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-import cz.xtf.core.bm.BuildManagers;
-import cz.xtf.core.openshift.OpenShift;
-import cz.xtf.core.openshift.OpenShifts;
-import cz.xtf.junit5.annotations.OpenShiftRecorder;
 import cz.xtf.junit5.config.JUnitConfig;
-import cz.xtf.junit5.extensions.helpers.EventsFilterBuilder;
-import cz.xtf.junit5.extensions.helpers.ResourcesFilterBuilder;
-import cz.xtf.junit5.extensions.helpers.ResourcesPrinterHelper;
-import cz.xtf.junit5.extensions.helpers.ResourcesTimestampHelper;
-import io.fabric8.kubernetes.api.model.ConfigMap;
-import io.fabric8.kubernetes.api.model.Event;
-import io.fabric8.kubernetes.api.model.HasMetadata;
-import io.fabric8.kubernetes.api.model.Pod;
-import io.fabric8.kubernetes.api.model.Secret;
-import io.fabric8.kubernetes.api.model.Service;
-import io.fabric8.kubernetes.api.model.apps.StatefulSet;
-import io.fabric8.openshift.api.model.Build;
-import io.fabric8.openshift.api.model.BuildConfig;
-import io.fabric8.openshift.api.model.DeploymentConfig;
-import io.fabric8.openshift.api.model.ImageStream;
-import io.fabric8.openshift.api.model.Route;
 
 /**
- * Record OpenShift isolated state relative to a test.
- * Resources are filtered by name provided via {@link OpenShiftRecorder} annotation. Names are turned into regexes
- * by adding {@code .*} as a suffix. If no name is provided, resources in namespaces (BM and master) are filtered automatically
- * by recording what resources are seen before test and so on.
- * <p>
- * Recorded resources:
- * <ul>
- * <li>pods states</li>
- * <li>deployment configs states</li>
- * <li>builds states</li>
- * <li>build configs states</li>
- * <li>image streams states</li>
- * <li>stateful sets states</li>
- * <li>routes states</li>
- * <li>services states</li>
- * <li>secrets states</li>
- * <li>logs of pods</li>
- * <li>events</li>
- * </ul>
- * <p>
- * OpenShift state is recorded when a test throws an exception. If {@link JUnitConfig#recordAlways()} is true, state is
- * recorded also when a test passes.
- * <p>
- * Use {@link JUnitConfig#recordDir()} ()} to set the directory of records.
+ * Implements JUnit lifecycle events handling in order to record OCP state for different purposes (e.g.: debug).
+ * The operational logic to perform the steps to get the status from OCP is delegated to
+ * {@link OpenShiftRecorderService}.
+ *
+ * Based on the type of the event that is being handled, an instance of this class will either call
```

Comment:
This paragraph is not accurate. Methods like `initFilters` are called always.

Reply:
Where? I see it was called on `beforeAll` before the refactoring and now it is similar to what was done before.
It doesn't seem not accurate to me, can you please elaborate on what you'd think should be here?

Reply:
I changed just the following lines, please see below.

Reply:
From the `type of the event` I get a feeling that only one of the methods mentioned later is called. `Accurate` might not have been the best wording from me. Maybe something like:
{@link OpenShiftRecorderService#initFilters(ExtensionContext)} - beforeAllCallback; used to get the info what resources are on a cluster before a test class
{@link OpenShiftRecorderService#updateFilters(ExtensionContext)} - beforeEachCallback; used to get the info what resources are on a cluster before a test
{@link OpenShiftRecorderService#recordState(ExtensionContext)} - on various events; record state of selected resources (by name) or make an estimation based on info from two steps above

Reply:
Thanks @istraka, I incorporated your suggestions.